src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestValueResolver.php

```diff
@@ -24,6 +25,17 @@ final class RequestValueResolver implements ValueResolverInterface
 {
     public function resolve(Request $request, ArgumentMetadata $argument): array
     {
-        return Request::class === $argument->getType() || is_subclass_of($argument->getType(), Request::class) ? [$request] : [];
+        $result = Request::class === $argument->getType() || is_subclass_of($argument->getType(), Request::class) ? [$request] : [];
+        if (!empty($result)) {
```

Comment:
I suggest writing the code like that:

```php
if (Request::class === $argument->getType() || is_subclass_of($argument->getType(), Request::class)) {
    return [$request];
}

// .. Same code below
```

It will be much easier to read than the current proposal checking whether the result is an empty array or no.

Reply:
Good point 👍